src/core/featuremodel.cpp

```diff
@@ -472,7 +472,22 @@ void FeatureModel::resetFeature()
   mFeature = QgsFeature( mLayer->fields() );
 }
 
-void FeatureModel::resetAttributes()
+void FeatureModel::resetFeatureId()
+{
+  if ( !mLayer )
+    return;
+
+  if ( sRememberings->contains( mLayer ) )
+  {
+    QMutex *mutex = sMutex;
```

Comment:
What's this copy for?

Reply:
@m-kuhn , the logic I had in mind is that whenever you call either resetFeature() or the newly introduced resetFeatureId(), the attribute values held in the mFeature at the time of those calls are remembered. It's not used here per say, but I think if we rely on this new function elsewhere, we'd likely want the remember logic to match that of resetFeature().

Reply:
I meant the mutex

Reply:
Oh sorry I thought you meant to code block -- you know what, I coded that forever ago and completely forgot why I did it this way throughout featuremodel.cpp :fearful: